plugins/quetz_sql_authenticator/quetz_sql_authenticator/cli.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+import click
+from passlib.hash import pbkdf2_sha256
+from sqlalchemy.exc import IntegrityError, NoResultFound
+from sqlmodel import Session, SQLModel, create_engine, select
+
+from .utils import Credentials
+
+
+def calculate_hash(value: str) -> str:
+    """Calculate hash from value."""
+    return pbkdf2_sha256.hash(value)
+
+
+@click.group()
+def _cli():
+    pass
+
+
+@_cli.command("create")
+@click.argument("username")
+@click.argument("password")
+@click.option(
+    "--database-url",
+    help="The SQLAlchemy URL pointing to the database.",
+    default=os.environ.get("QUETZ_SQL_AUTHENTICATOR_DATABASE_URL"),
+)
+def _create(username: str, password: str, database_url: str) -> None:
+    credentials = Credentials(username=username, password=calculate_hash(password))
+    with Session(create_engine(database_url)) as session:
+        try:
+            session.add(credentials)
+            session.commit()
+        except IntegrityError:
+            click.echo(f"ERROR: User '{username}' already exists.")
```

Comment:
Could there be other integrity errors? Maybe it's better to print the database's message here.

Reply:
good point, there could be other errors. i could also remove the except block altogether and just raise hard, whenever something fails.